addons/wowl/static/src/components/dropdown/dropdown.ts

```diff
@@ -0,0 +1,124 @@
+import { Component, core, hooks, useState } from "@odoo/owl";
+import { OwlEvent } from "@odoo/owl/dist/types/core/owl_event";
+import { DropdownItemClickedPayload, ParentClosingMode } from "./dropdown_item";
+
+export class Dropdown extends Component {
+  static bus = new core.EventBus();
+  static props = {
+    open: {
+      type: Boolean,
+      optional: true,
+    },
+    menuClass: {
+      type: String,
+      optional: true,
+    },
+    togglerClass: {
+      type: String,
+      optional: true,
+    },
+  };
+  static template = "wowl.Dropdown";
+
+  state = useState({ open: this.props.open, groupIsOpen: false });
```

Comment:
I had the same idea, but did not dig into it.

Why ? Because I can accept that the Dropdown class has more logic that we originally thought, as long as the user API is simplified. And here it is simpler. Off course, there is more magic happening behind the scenes, but LBH it is not so bad and not so difficult to maintain.

I do not see a DropdownOrchestrator bringing so much advantages, while I can see it would bring a less simpler API.

These are just my thoughts :-)